### DISCUSSION TITLE
Default colors of FileSystemInfo objects

### DISCUSSION THREAD

**Discussion by SteveL-MSFT** [PowerShell/PowerShell#15948] [Enhancements]
Currently, the default colors are largely derived from `ls --color`.  However, folders didn't have sufficient contrast being just blue, so that was white on blue to improve accessibility.  The PowerShell files are yellow to make them standout.  Open to suggestions on the default colors as well as any extra default extensions (but let's not go crazy here).

**Comment by ChrisWarwick** [comment]
Quite a large proportion of the population is colour blind. Can you please ensure the colour mix is testing by someone unfortunate enough to have, at least the most common forms, of this affliction? The results may appear somewhat garish to those of you who are lucky enough to make out subtle shades - sorry!

**Reply by theJasonHelmick** [reply]
Good point Chris. We will get the accessibility folks to have a look and we will do our best understanding that this will effect people differently.  Please continue to help us by commenting.

**Comment by jhoneill** [comment]
First off. When this is enabled, if someone runs pwsh on windows from outside windows terminal a bug in the old console host's handling of ansi means you can't scroll up and down a directory listing.  https://github.com/PowerShell/PowerShell/issues/15945#issuecomment-901297376    **To me**, unless that works - even through the break is outside of PowerShell - the feature needs to stay experimental "I ran pwsh and can't reliably see a list of files" isn't a user experience anyone wants, and "Run it in terminal if you want scroll to work" won't do as a response  

Secondly to unify what I said  elsewhere https://github.com/PowerShell/PowerShell/pull/14403#issuecomment-901068914  here is a list of my problems : I *do* understand that I might represent a certain subset of people with undiagnosed accessibility issues, and therefore using something other than the defaults might be appropriate because I'm not the majority. But here goes: 

![image](https://user-images.githubusercontent.com/16209593/129890706-c06158a1-68b3-43c6-84c2-e20c9b17b02a.png)

1. The green header "Splits" the screen and "orphans" the directory name.  Bold or brighter text, makes sense for a header,, but who puts the header rows in their Excel or Word documents in a different colour ? If we are going to have that then directory name is part of the header and should be in the accent colour, but `$PSStyle.Formatting.TableHeader` doesn't do that.
2. The Directories are URGENT. I know they are directories because (a) they are sorted to the top and (b) there is a blank space where the length would go. It's pulling my eye to something, that I don't need to know.  Cyan instead of blue if they must be coloured at all. 
3. I also know the symbolic link is a link because it has the target next to it (now in a different colour).  Because we can see where it links to a symlink doesn't need to be different from directories.  
4. What is dangerous or erroneous about ZIP files?  If there is a need to distinguish them at all (questionable) **should they be shouting for attention ?**  red is a bad choice (c.f. Get-Error where the use of Red means the error text jumps out from everything else - which _is_ useful)
5. It may be my eyesight, but Green "recedes" so `less.exe`  is harder to pick out, especially when sandwiched between _emphasized_ PS1 files.  (Although it says it is foreground 32 - Green, it appears to render as light green 92 so I couldn't make it more prominent. )
6. Emphasizing PowerShell files has my brain asking "why are these different from those"  and colouring .psd  and .psxml files but  not json and xml files compounds the problem. 

The whole idea of a user being able to say "I want this set of files highlighted in one colour, and this other set in another colour" is good. But PS files in yellow and archives in red... doesn't make any sense ? 
Executables in green makes sense on Linux but on Windows where the extension tells us... not so much. 

The net effect of 1-6 is that it takes me longer to find what I'm looking for in a listing.   

I've simplified it to  
```
$PSStyle.Formatting.TableHeader = "`e[97;1m"
$PSStyle.FileInfo.Directory = $PSStyle.FileInfo.SymbolicLink
$PSStyle.FileInfo.Extension.Keys | % {$PSStyle.FileInfo.Extension.Remove($_)}
$PSStyle.FileInfo.Extension.Add(".ps1",$PSStyle.FileInfo.Executable)
```
![image](https://user-images.githubusercontent.com/16209593/129955151-1ba73969-cfa7-4bf4-9bad-050391e1cbd8.png)

So basically the colours are directories, things that run, things that don't. The header is brightened. 


When I might want colour to give me more information , e.g. intellisense with [ctrl]+[space] there is none. 
![image](https://user-images.githubusercontent.com/16209593/129889083-bf8ddc3e-5191-4c91-8a80-ef56774cc84e.png)

Here,  picking out directories with colour or underline would help, so might a warning I'm about to move an exe or a script whose path is hard coded  somewhere.  But it's not available, and if it were it would probably require a listing to use the same colours, making it all the more important to get good defaults.

**Comment by theJasonHelmick** [comment]
@jhoneill - On your first point - yup, maybe we can disable colors for the console as I don't know what the appetite is to address this - however, on your second point regarding the colors - would being able to set and select a theme help here?  So, Steve and I may like certain colors, and they may meet our accessibility requirements, but would they meet everyone's desire? No.  While the colors are kinda simple to set, I wonder if being able to choose a theme would be a better customer experience.  That said, the defaults we ship are important, even with theme settings, so I appreciate your feedback.

**Reply by jhoneill** [reply]
@theJasonHelmick 
On breaking, if you can't get a directory listing that scrolls up and down in the default host , it's probably not ready for prime time. 

On the second, the more time I spend with this the more I think themes will be the answer. "A" may like something "B" thinks is garish and "A" might see B's choice as drab. "C" might need something which will print in black and white. Then there are accessibility needs... Person D comes along and says "Hey that would work for me", but it's only sharable by adding a bunch of lines to a profile... 

I'm starting to classify myself as having accessibility issues - I cannot read text when there is something moving or flashing elsewhere on the screen - I need to put my hand over it (ad-blockers make some sites usable).  VS Code is better for me with most of the floaters removed. I'd expected the suggested text you worked on would be a problem, but it works for me with the text being quite faint and saves me a few seconds dozens of times a day, In something like Get-Error, making things jump out helps me as the author intended.  But 
when I'm confronted by output with lots of changes of colour and/or reversed text (like the default file colours) I'm very aware that my eye no longer scans smoothly down the column. With the column headings emphasized as "Start of output", my brain edits out the directory name. 

There is also no agreement on what the ansi-colors _are_  you would think "White" and "bright white" are constant.  But you can see below the defaults in Windows Terminal and VS Code switch the meaning of 37 and 97. You can also see yellow is noticeably different. 
![image](https://user-images.githubusercontent.com/16209593/130758649-de92a71b-38a8-46b2-8e6f-1a1532b08d22.png)


Terminal 
![image](https://user-images.githubusercontent.com/16209593/130760121-422cf702-9c11-4e31-bf07-69f77b70f22c.png)
VsCode
![image](https://user-images.githubusercontent.com/16209593/130760271-c111c58f-9737-4f91-afaa-cffd0e0ea0b0.png)

But each _theme_ in VSCode  and each _color scheme_ Terminal can redefine any ansi-color. I'm using the PSISE scheme in code which doesn't set anything for the terminal, and only my background is changed the default "Campbell" scheme in terminal.  With these you can see @SteveL-MSFT 's concern with blue - bright blue [94] has enough contrast in terminal, but doesn't in VSCode.   Code is better at sharing it's themes than terminal is.    
So themes for PSStyle wouldn't be _universal_  I'm very wary of having a default theme which is just what seemed like a good idea to the person who coded the feature.
